### PR TITLE
feature/boas 731 remove documents from queue

### PR DESCRIPTION
**Pull request by beejjacobs**
## Describe your changes

1. Changes to a documents published status now save the previous status to the database.
2. A new API endpoint for reverting the published status has been added
3. The front-end calls revert status when a user clicks "Remove" from the publishing queue.

## Issue ticket number and link

BOAS-731
https://pins-ds.atlassian.net/jira/software/projects/BOAS/boards/172?selectedIssue=BOAS-731

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by mayya-bond-kc — COMMENTED** [review]
Looks good overall. I have made some comments on the styling and a case where I think the error you are hoping to throw might not be thrown. Good to double check this. As well, Could we please add some tests for the new endpoint that will revert the document status?

File: apps/api/src/server/applications/application/documents/document.controller.js [hunk 1]
```diff
@@ -126,6 +155,39 @@ export const getDocumentProperties = async ({ params: { id: caseId, guid } }, re
...
 
+/**
+ * Revert the published status of a document to the previous published status.
+ *
+ * @type {import('express').RequestHandler<{id: number;guid: string}, any, any, any>}
+ */
+export const revertDocumentPublishedStatus = async ({ params: { guid } }, response) => {
+	const documentVersion = await documentVersionRepository.getById(guid);
+
+	if (!documentVersion) {
```
[the remark this answers is not in the document]
Comment (beejjacobs) [reply apps/api/src/server/applications/application/documents/document.controller.js:166]: Happy to have a more specific check, e.g. `documentVersion === null`, but the currently logic should work as expected if its null, as the `if` statement is true, so the error is thrown. I'll add a test for that.

File: apps/api/src/server/applications/application/documents/document.controller.js [hunk 2]
```diff
@@ -126,6 +155,39 @@ export const getDocumentProperties = async ({ params: { id: caseId, guid } }, re
...
+	let publishedStatus;
+
+	if (
+		typeof documentVersion.publishedStatusPrev !== 'undefined' &&
+		documentVersion.publishedStatusPrev !== null
+	) {
+		publishedStatus = documentVersion.publishedStatusPrev;
+	} else {
+		throw new BackOfficeAppError(`No previous published status to revert to`, 412);
+	}
```
[the remark this answers is not in the document]
Comment (beejjacobs) [reply apps/api/src/server/applications/application/documents/document.controller.js]: Yup that's fair, it'll make the flow of logic clearer: first precondition checks, then implementation.

Updated: https://github.com/Planning-Inspectorate/back-office/pull/854/commits/e6b95cb558224c5cc07e21c4fee6845970ef8513

File: apps/api/src/server/applications/application/documents/document.controller.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (beejjacobs) [reply apps/api/src/server/applications/application/documents/document.controller.js:166]: Added a test for this case: https://github.com/Planning-Inspectorate/back-office/pull/854/commits/0cfe30d21bb8cc0af013507fda384090f91a5641

[the remark this answers is not in the document]
Comment (mayya-bond-kc) [reply apps/api/src/server/applications/application/documents/document.controller.js:166]: Ah super, thanks for double checking this and adding the tests!

**Review by mayya-bond-kc — APPROVED** [review]

**Pull request merged**